### PR TITLE
Support rfc1123

### LINKED ISSUE #23: rfc1123: labels can have initial digits - regression  on master
1123 authoritatively relaxes 1035

this is a regression on master

per #22

### PR DESCRIPTION
**Pull request by wakemaster39**
Resolves #23

### CODE REVIEW

**Comment by ypcrts** [comment]
LGTM but y tests no pass

**Comment by wakemaster39** [comment]
@ypcrts test pass, linting failed.

I setup all my projects with black having a line length of 120 instead of the default 80 and didn;t think about how it was impacting this project. 80 to me is just crazy small, even linux kernel has 100 length now. I can resubmit with it down to 80, or I can alternative submit with black configured for a longer line length as well.

Let me know.

**Comment by ypcrts** [comment]
No problem. I’ll merge this today

**Comment by wakemaster39** [comment]
@ypcrts just checking in on this.

File: setup.py [hunk 1]
```diff
@@ -5,16 +5,15 @@
...
+        "Validate fully-qualified domain names compliant " "to RFC 1035 and the preferred form in RFC 3686 s. 2."
     ),
     author="ypcrts",
     author_email="ypcrts@users.noreply.github.com",
     url="https://github.com/ypcrts/fqdn",
     keywords=["fqdn", "domain", "hostname", "RFC3686", "dns"],
     license="MPL 2.0",
     zip_safe=True,
-    install_requires=["cached-property>=1.3.0"],
+    install_requires=["cached-property>=1.3.0;python_version<'3.8'"],
```
Comment (ypcrts) [inline setup.py:16]: thx

File: tests/test_fqdn.py [hunk 2]
```diff
@@ -72,52 +71,36 @@ def test_rfc_3696_s_2__label_invalid_starts_or_ends_with_hyphen(self):
...
+        self.__assert_valid_fwd_and_bkwd_from_seq("sh4d05-7357", "c00-mm", allow_underscores=a_u)
 
-    def test_rfc_1035_s_2_3_1__label_cannot_have_inital_digit(self, a_u):
-        self.__assert_invalid_fwd_and_bkwd_from_seq("www", "1", allow_underscores=a_u)
-        self.__assert_invalid_fwd_and_bkwd_from_seq("1w", "1", allow_underscores=a_u)
-        self.__assert_invalid_fwd_and_bkwd_from_seq("1w", "a", allow_underscores=a_u)
-        self.__assert_invalid_fwd_and_bkwd_from_seq("1w1", "d", allow_underscores=a_u)
-        self.__assert_invalid_fwd_and_bkwd_from_seq("111", "a", allow_underscores=a_u)
-        self.__assert_invalid_fwd_and_bkwd_from_seq("www", "1a", allow_underscores=a_u)
+    def test_rfc_1123_s_2_3_1__label_can_have_inital_digit(self, a_u):
```
Comment (ypcrts) [inline tests/test_fqdn.py:95]: ✔️

File: setup.py [hunk 3]
```diff
@@ -5,16 +5,15 @@
     packages=["fqdn"],
     version="1.4.0",
     description=(
-        "Validate fully-qualified domain names compliant "
-        "to RFC 1035 and the preferred form in RFC 3686 s. 2."
+        "Validate fully-qualified domain names compliant " "to RFC 1035 and the preferred form in RFC 3686 s. 2."
```
Comment (ypcrts) [inline setup.py:8]: TODO: i need to fix this

File: tests/test_fqdn.py [hunk 4]
```diff
@@ -72,52 +71,36 @@ def test_rfc_3696_s_2__label_invalid_starts_or_ends_with_hyphen(self):
...
+        self.__assert_valid_fwd_and_bkwd_from_seq("www", "1", allow_underscores=a_u)
+        self.__assert_valid_fwd_and_bkwd_from_seq("1w", "1", allow_underscores=a_u)
+        self.__assert_valid_fwd_and_bkwd_from_seq("1w", "a", allow_underscores=a_u)
+        self.__assert_valid_fwd_and_bkwd_from_seq("1w1", "d", allow_underscores=a_u)
+        self.__assert_valid_fwd_and_bkwd_from_seq("111", "a", allow_underscores=a_u)
+        self.__assert_valid_fwd_and_bkwd_from_seq("www", "1a", allow_underscores=a_u)
 
-    def test_rfc_1035_s_2_3_1__label_can_have_medial_and_terminal_digits(self, a_u):
+    def test_rfc_1123_s_2_3_1__label_can_have_medial_and_terminal_digits(self, a_u):
```
Comment (ypcrts) [inline tests/test_fqdn.py:103]: TODO: i will rever this line change

**Review by ypcrts — APPROVED** [review]

**Comment by ypcrts** [comment]
thanks again

**Comment by ypcrts** [comment]
resolves #24

**Comment by ypcrts** [comment]
resolves #22

**Pull request merged**